pkg/operation/botanist/controlplane.go

```diff
@@ -1128,10 +1169,86 @@ func (b *Botanist) DeployETCD(ctx context.Context) error {
 				}
 			}
 		}
-		if err := b.ApplyChartSeed(filepath.Join(chartPathControlPlane, "etcd"), b.Shoot.SeedNamespace, fmt.Sprintf("etcd-%s", role), nil, etcd); err != nil {
+		if err := b.ApplyChartSeed(filepath.Join(chartPathControlPlane, "etcd"), b.Shoot.SeedNamespace, fmt.Sprintf("etcd-%s", role), nil, values); err != nil {
 			return err
 		}
 	}
 
 	return nil
 }
+
+// DetermineBackupSchedule determines the backup schedule based on the shoot creation and maintenance time window.
+func DetermineBackupSchedule(shoot *gardencorev1beta1.Shoot, etcd *druidv1alpha1.Etcd) (string, error) {
+	var (
+		begin, end string
+		shootUID   types.UID
+	)
+
+	if etcd.Spec.Backup.FullSnapshotSchedule != nil {
+		return *etcd.Spec.Backup.FullSnapshotSchedule, nil
+	}
+
+	if shoot.Spec.Maintenance != nil && shoot.Spec.Maintenance.TimeWindow != nil {
+		begin = shoot.Spec.Maintenance.TimeWindow.Begin
+		end = shoot.Spec.Maintenance.TimeWindow.End
+		shootUID = shoot.Status.UID
+	}
+
+	if len(begin) != 0 && len(end) != 0 {
+		maintenanceTimeWindow, err := utils.ParseMaintenanceTimeWindow(begin, end)
+		if err != nil {
+			return "", err
+		}
+
+		if !maintenanceTimeWindow.Equal(utils.AlwaysTimeWindow) {
+			// Randomize the snapshot timing daily but within last hour.
+			// The 15 minutes buffer is set to snapshot upload time before actual maintenance window start.
+			snapshotWindowBegin := maintenanceTimeWindow.Begin().Add(-1, -15, 0)
+			randomMinutes := int(crc32.ChecksumIEEE([]byte(shootUID)) % 60)
+			snapshotTime := snapshotWindowBegin.Add(0, randomMinutes, 0)
+			return fmt.Sprintf("%d %d * * *", snapshotTime.Minute(), snapshotTime.Hour()), nil
+		}
+	}
+
+	creationMinute := shoot.CreationTimestamp.Minute()
+	creationHour := shoot.CreationTimestamp.Hour()
+	return fmt.Sprintf("%d %d * * *", creationMinute, creationHour), nil
+}
+
+// DetermineDefragmentSchedule determines the defragment schedule based on the shoot creation and maintenance time window.
+func DetermineDefragmentSchedule(shoot *gardencorev1beta1.Shoot, etcd *druidv1alpha1.Etcd) (string, error) {
+	var (
+		begin, end string
+		shootUID   types.UID
+	)
+
+	if etcd.Spec.Etcd.DefragmentationSchedule != nil {
+		return *etcd.Spec.Etcd.DefragmentationSchedule, nil
+	}
+
+	if shoot.Spec.Maintenance != nil && shoot.Spec.Maintenance.TimeWindow != nil {
+		begin = shoot.Spec.Maintenance.TimeWindow.Begin
+		end = shoot.Spec.Maintenance.TimeWindow.End
+		shootUID = shoot.Status.UID
+	}
+
+	if len(begin) != 0 && len(end) != 0 {
+		maintenanceTimeWindow, err := utils.ParseMaintenanceTimeWindow(begin, end)
+		if err != nil {
+			return "", err
+		}
+
+		if !maintenanceTimeWindow.Equal(utils.AlwaysTimeWindow) {
```

Comment:
For one, full snapshot random time should be accommodated -1.30 hrs prior to maintenance window. For defrag schedule, random time should be from maintenance window start. The cron strings is also different. I think it would be futile to pull out common lines of code here.